src/service/viewer-impl.js

```diff
@@ -544,7 +544,7 @@ export class Viewer {
     const canonicalUrl = Services.documentInfoForDoc(this.ampdoc).canonicalUrl;
     const canonicalSourceOrigin = getSourceOrigin(canonicalUrl);
     if (this.hasRoughlySameOrigin_(sourceOrigin, canonicalSourceOrigin)) {
-      const oldFragment = getFragment(this.win.location.href);
+      const oldFragment = getFragment(this.win.location.href || this.win.location.originalHash);
```

Comment:
Just use the `originalHash`, `location.href` is unnecessary.

Reply:
Thanks for the lightning fast review, Justin!

If I only use the originalHash, the viewer-test.js tests fail (presumably because the original hash isn't being coped over to originalHash, so it's undefined, causing getFragment to fail).

Should I update my tests to explicitly set originalHash instead of href? If so, is it safe to assume that in production originalHash will *always* be set by the time maybeUpdateFragmentForCct is called?

Reply:
I think it's fine as is. A lot of places internally do this -- I'd just flip the order around.